server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java

```diff
@@ -399,14 +403,35 @@ private Locality retrievePosition(@NonNull ParserRuleContext ctx) {
 
   @NonNull
   private Function<Token, Locality> toPosition() {
-    return token ->
-        Locality.builder()
-            .uri(documentUri)
-            .copybookId(retrieveCopybookId())
-            .range(new Range(getStartPosition(token), getStopPosition(token)))
-            .token(token.getText())
-            .recognizer(GrammarPreprocessorListenerImpl.class)
-            .build();
+    return token -> {
+      Position startPosition;
+      Position stopPosition;
+      if (StringUtils.isNotBlank(token.getText())
```

Comment:
optional: can be extracted to a validation method, regarding to positionCorrectionMap

Reply:
done.